src/main/java/hudson/scm/CVSChangeLogSet.java

```diff
@@ -179,23 +177,19 @@ public static CVSChangeLogSet parse(final Run<?, ?> build, RepositoryBrowser<?>
         private static final DateFormat[] dateFormatters = new SimpleDateFormat[]{
             new SimpleDateFormat(CHANGE_DATE_FORMATTER_PATTERN),
             new SimpleDateFormat("yyyy/MM/dd HH:mm:ss")};
-        private static final DateFormat DATE_FORMATTER = new SimpleDateFormat(
-                "yyyy-MM-dd");
-        private static final DateFormat TIME_FORMATTER = new SimpleDateFormat(
-                "HH:mm");
 
         private String user;
         private String msg;
         private final List<File> files = new ArrayList<File>();
-        private Calendar changeDate;
+        private long changeDate;
         private CvsRepository repository;
 
         /**
          * Returns true if all the fields that are supposed to be non-null is
          * present. This is used to make sure the XML file was correct.
          */
         public boolean isComplete() {
-            return changeDate != null && msg != null;
+            return changeDate != 0 && msg != null;
```

Comment:
"0" is a default value if the field is not present. It may cause the  "complete" entries to switch to "incomplete" aftr reloading from the disk. Not sure what would be the impact, but it smells

Reply:
Not sure what you think the issue would be. The logic is analogous to when the field was of a reference type. It is either initialized or not.